patroni/postgresql/slots.py

```diff
@@ -192,10 +192,19 @@ def ignore_replication_slot(self, cluster, name):
         return False
 
     def drop_replication_slot(self, name):
-        cursor = self._query(('SELECT pg_catalog.pg_drop_replication_slot(%s) WHERE EXISTS (SELECT 1 ' +
-                              'FROM pg_catalog.pg_replication_slots WHERE slot_name = %s AND NOT active)'), name, name)
-        # In normal situation rowcount should be 1, otherwise either slot doesn't exists or it is still active
-        return cursor.rowcount == 1
+        # First check if the replication slot is still active; not being able
+        # to drop it in this case is not considered an error
+        cursor_slot_active = self._query(('SELECT 1 FROM pg_catalog.pg_replication_slots WHERE ' +
+                                          'slot_name = %s AND active'), name)
+        if (cursor_slot_active.rowcount == 1):
+            logger.warning("Unable to drop unknown replication slot '%s', slot is still active", name)
+            return True
```

Comment:
Since the method returns True from here there will be no further attempt to remove the slot.

Reply:
oh right I see - if the slots becomes inactive, we'd never drop it.

So let's get back to logging this at debug each cycle, will revise

Reply:
I rearranged this now, I hope this is better